### PR TITLE
Add new user demo to community page

### LINKED ISSUE #219: [DEMO] QCNN for Speech Recognition and Neural Saliency Analysis
#### General information

**Name**
C.-H. Huck Yang

**Affiliation** 
Georgia Institute of Technology, Atlanta, GA

--------------------------------------------------------------------------------

#### Demo information

**Title**
QCNN for Speech Commands Recognition.

**Abstract**
We provide a hybrid model training with larger-scale acoustic features 3,000 to 10,000 with quantum convolution networks with Random layer, which still provide insightful convolution features without the encoding time cost from the CPU simulation and on the request queueing time from QPU. We further provide classical activation mapping, a neural saliency analysis, on the well-trained neural models (QCNN Self-Attention v.s. CNN Self-Attention) to justify that the QCNN Self-attention model did learn meaningful representations. An additional Connectionist Temporal Classification (CTC) loss on character recognition is also provided for continuous speech recognition. 

**Relevant links**

We provide a GitHub [repo](https://github.com/huckiyang/QuantumSpeech-QCNN) and [Colab](https://colab.research.google.com/drive/11Yi53W6-Z-uW84A8Sr6OqOMlBIel1PKM?usp=sharing) for future studies. A related [preprint](https://arxiv.org/pdf/2010.13309.pdf) has been released and will appear in IEEE International Conference on Acoustics, Speech, & Signal Processing ([ICASSP](https://scholar.google.com/citations?view_op=top_venues&hl=en&venue=HHC6AUo36fEJ.2020&vq=phy_acousticssound)) 2021.

```bib
@article{yang2020decentralizing,
  title={Decentralizing Feature Extraction with Quantum Convolutional Neural Network for Automatic Speech Recognition},
  author={Yang, Chao-Han Huck and Qi, Jun and Chen, Samuel Yen-Chi and Chen, Pin-Yu and Siniscalchi, Sabato Marco and Ma, Xiaoli and Lee, Chin-Hui},
  journal={arXiv preprint arXiv:2010.13309},
  year={2020}
}
```

### PR DESCRIPTION
**Pull request by agran2018**
Add the user demo in #219 to the community page

Closes #219

### CODE REVIEW

**Comment by co9olguy** [comment]
Oops, looks like we've got too many cards in the first row
image

**Comment by trbromley** [comment]
Yes, we need to fix the cards. A side-effect of https://github.com/PennyLaneAI/qml/pull/212 is that it's now nontrivial to insert a demo so that there's overall an odd number of demos (as is the case here). We can either:
- Introduce a new row at the top with the demo and an empty slot
- Insert the demo and bump all other demos down by one. Sounds easy but I think it involves a lot of reshuffling.

Since we have 3 demos to merge in this week (also resulting in an overall odd number), I'd recommend here adding a new row and an empty slot and merging all the 3 demos roughly at the same time with a final reshuffle.

**Comment by co9olguy** [comment]
@josh146 could we use a template to programmatically populate the cards, avoiding the need for hand-tuning every time there's a new entry?

**Comment by josh146** [comment]
> @josh146 could we use a template to programmatically populate the cards, avoiding the need for hand-tuning every time there's a new entry?

Great idea! We could have the following setup:

* A json/yaml file that contains the demo card data:

  ```yaml
  - demo
    - title: asdkljasdlk;jsad
    - author: adfk;jhasdf
    - date: ...
  ```

* A Jinja template for the ReST file

* A Python script that parses the json/yaml, and fills the ReST template for Sphinx to then use.

**Comment by agran2018** [comment]
In line 50 it reads:

> .. For two cards per row, create a new card deck every two cards.

should we do this in the meantime?

File: demos_community.rst [hunk 1]
```diff
@@ -58,6 +58,23 @@ Community
...
+
+    We provide a hybrid model training with larger-scale acoustic features 3000 to 10000 with
+    quantum convolution neural networks (QCNN) with Random layer, which still provide insightful
+    convolution features without the encoding time cost from the CPU simulation and on the request
+    queueing time from QPU. We further provide classical activation mapping, a neural saliency
+    analysis, on the well-trained neural models (QCNN Self-Attention vs. CNN Self-Attention) to
+    show that the QCNN Self-attention model did learn meaningful representations. An additional
+    Connectionist Temporal Classification (CTC) loss on character recognition is also provided for
+    continuous speech recognition.
+
```
Comment (trbromley) [inline demos_community.rst:77]:
```suggestion
:html:`<div class="card hidden-card"></div></div>`

:html:`</div></div><br><div style='clear:both'>`
:html:`<div class="card-deck">`
```
This should work, but we should check how it renders.

**Review by trbromley — APPROVED** [review]

**Comment by co9olguy** [comment]
@trbromley :sob: 
image

**Comment by agran2018** [comment]
Fixed.

Screenshot from 2021-03-03 18-59-33

**Comment by trbromley** [comment]
Let's wait until https://github.com/PennyLaneAI/qml/pull/225 is merged before merging this.

**Comment by co9olguy** [comment]
#225 has now been merged, so good to port this one over now

**Comment by agran2018** [comment]
> #225 has now been merged, so good to port this one over now

Done. Works great now. I also checked that the links to the code and the reference paper. 
Screenshot from 2021-03-05 13-12-12

**Comment by trbromley** [comment]
Thanks @agran2018!

**Pull request merged**